rpcd/playbooks/stage-python-artifacts.yml

```diff
@@ -0,0 +1,184 @@
+    https_proxy_env: "{{ lookup('env', 'https_proxy') | default('not_set', true) }}"
+  tasks:
+
+    - name: Set the staging path
+      set_fact:
+        staging_path: |-
+          {%- if groups['repo_all'] is defined -%}
+          /openstack/{{ hostvars[groups['repo_all'][0]]['inventory_hostname'] }}/repo
+          {%- else -%}
+          /openstack/stage
+          {%- endif -%}
+      tags:
+        - always
+
+    - name: Fetch the upstream manifest file
+      uri:
+        url: "{{ upstream_artifact_base_url }}/os-releases/{{ rpc_release }}/MANIFEST.in"
+        return_content: yes
+        validate_certs: yes
+      register: manifest
+      tags:
+        - always
+
+    - name: Fetch the upstream venv list
+      uri:
+        url: "{{ upstream_artifact_base_url }}/venvs/{{ rpc_release }}/ubuntu/"
+        return_content: yes
+        validate_certs: yes
+      register: venv_index
+      tags:
+        - always
+
+    - name: Derive the lists of artifacts to download
+      set_fact:
+        python_wheel_list: |
+          {%- set content_list = manifest.content.split('\n') -%}
+          {%- set result_list = [] -%}
+          {%- for item in content_list -%}
+          {%-   if item.split('/')[0] == 'pools' -%}
+          {%-     set _ = result_list.append(item) -%}
+          {%-   endif -%}
+          {%- endfor -%}
+          {{- result_list -}}
+        git_folder_list: |
+          {%- set content_list = manifest.content.split('\n') -%}
+          {%- set result_list = [] -%}
+          {%- for item in content_list -%}
+          {%-   if item.split('/')[0] == 'openstackgit' -%}
+          {%-     set _ = result_list.append(item) -%}
+          {%-   endif -%}
+          {%- endfor -%}
+          {{- result_list -}}
+        python_venv_list: |
+          {%- set content_list = venv_index.content.split('\r\n') -%}
+          {%- set result_list = [] -%}
+          {%- for item in content_list -%}
+          {%-   if item | search("<a href.*>.*</a>") -%}
+          {%-     set item_clean = item | regex_replace(".*<a href.*>(.*)</a>.*", "\\1") -%}
+          {%-     if item_clean != "../" -%}
+          {%-       set _ = result_list.append(item_clean) -%}
+          {%-     endif -%}
+          {%-   endif -%}
+          {%- endfor -%}
+          {{- result_list -}}
+      tags:
+        - always
+
+    - name: Git staging folders setup
+      file:
+        path: "{{ item }}"
+        state: "directory"
+      with_items:
+        - "{{ staging_path }}/links"
+        - "{{ staging_path }}/openstackgit"
+        - "{{ staging_path }}/os-releases/{{ rpc_release }}"
+        - "{{ staging_path }}/venvs/{{ rpc_release }}/ubuntu"
+      tags:
+        - always
+
+    - name: Clone git repositories asynchronously
+      git:
+        repo: "{{ upstream_artifact_base_url }}/{{ item }}"
+        dest: "{{ staging_path }}/{{ item }}"
+        version: master
```

Comment:
Unfortunately not setting a version, or setting it to HEAD causes ansible-lint to fail. Ultimately the repo checkout doesn't matter because the only tasks that use these git sources after the wheels are built are the novnc/spice tasks and they do their own version checkout.